src/main/java/com/mcneilio/orcmaker/BasicEventDriver2.java

```diff
@@ -0,0 +1,149 @@
+package com.mcneilio.orcmaker;
+
+import com.mcneilio.orcmaker.columns.ColumnInitializer;
+import com.mcneilio.orcmaker.metrics.MetricsClient;
+import com.mcneilio.orcmaker.orcer.Orcer;
+import com.mcneilio.orcmaker.storage.StorageDriver;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.vector.ColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.LongColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.orc.OrcFile;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.Writer;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Properties;
+import java.util.UUID;
+
+
+public class BasicEventDriver2 implements EventDriver {
+
+  private final MetricsClient metrics;
+
+  Orcer[] orcers;
+
+  String[] colNames;
+  VectorizedRowBatch batch;
+  TypeDescription schema;
+  HashMap<String, ColumnVector> columns;
+  String eventName;
+  String eventDate;
+  String fileName;
+  String statsdEnv;
+  Configuration conf = new Configuration();
+  Writer writer = null;
+  StorageDriver storageDriver;
+  ColumnVector[] columnVectors;
+
+
+  public BasicEventDriver2(Properties config, TypeDescription typeDescription, StorageDriver storageDriver) {
```

Comment:
Could we just create a `mock` storageDriver implementation to allow configuring mocks through the props?

Reply:
I mean maybe we do this, but also nevermind, I added support for dependency injection in `BasicEventDriver`